extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -467,14 +469,13 @@ private PkceStateBean createPkceStateBean(TenantConfigContext configContext) {
                             return Uni.createFrom().failure(new AuthenticationCompletionException(tOuter));
                         }
 
-                        boolean internalIdToken = false;
+                        boolean internalIdToken = !configContext.oidcConfig.authentication.isIdTokenRequired().orElse(true);
                         if (tokens.getIdToken() == null) {
-                            if (configContext.oidcConfig.authentication.isIdTokenRequired().orElse(true)) {
+                            if (!internalIdToken) {
                                 return Uni.createFrom()
                                         .failure(new AuthenticationCompletionException("ID Token is not available"));
                             } else {
-                                tokens.setIdToken(generateInternalIdToken(configContext.oidcConfig));
-                                internalIdToken = true;
+                                tokens.setIdToken(generateInternalIdToken(configContext.oidcConfig, null));
```

Comment:
The 2nd one will override the first one if userinfo is required to be inlined

Reply:
In this case IdentityProvider already needs to have some token in case the cache is required, this is why the one without UserInfo is still generated, and then it is replaced once IdentityProvider completes its work, it is a little bit suboptimal, but having only a single internal token generation in this case would require passing some indicators via routing context, adding more ifs/else in `CodeAuthenticationMechanism` and `OidcIdentityProvider` and would be messy. The internal IdToken is very light weight, so regenerating with a symmetric signature it is not expensive so should be fine